### PR TITLE
fix(NavigationBar): preserve sequential mobile Tab order

### LINKED ISSUE #1070: NavigationBar: make pending and ready mobile Tab boundaries agree
Parent: #1046

## Problem

Two mobile NavigationBar Tab paths disagree with the actual sequential focus order:

1. With `menuTogglePlacement="before-brand"`, Tab pressed before the first portaled-panel position completes always picks the first navigation item instead of the same brand-first destination used once positioning is ready.
2. Panel boundary detection treats every enabled `NavigationItem` as a Tab boundary even when it is not sequentially tabbable (for example `tabindex="-1"`), so forward or reverse Tab can escape from the wrong item.

## Required behavior

- Pending and position-ready focus resolution use one ordering model.
- Brand-first placement produces the same Tab destination before and after positioning settles.
- Boundary detection uses the actual sequentially tabbable navigation items, excluding enabled-but-untabbable items.
- Forward and reverse bridging preserve the adjacent page focus order.

## Acceptance criteria

- Add regressions for Tab during initial pending positioning with `before-brand`.
- Add regressions where first or last enabled items have `tabindex="-1"`.
- Focused NavigationBar tests, component artifact checks, package typecheck, and relevant lint pass.
- Add a patch changeset.

This covers items 1 and 4 from #1046. It is distinct from #1048, which tracks completeness of the shared external-control tabbability helper.

### PR DESCRIPTION
**Pull request by stevekinney**
Closes #1070

- preserve the before-brand toggle destination while mobile panel positioning is pending
- use the existing sequential focus model for navigation-item Tab boundaries
- add regression coverage and document the keyboard behavior

Validation:
- bun run --filter=@lostgradient/cinder test -- src/components/navigation-bar/navigation-bar.test.ts
- bun run --filter=@lostgradient/cinder typecheck
- bun run --filter=@lostgradient/cinder lint
- bun run --filter=@lostgradient/cinder components:check
- bun run --filter=@lostgradient/cinder exports:check

### CODE REVIEW

File: packages/components/src/components/navigation-bar/navigation-bar.test.ts [hunk 1]
```diff
@@ -935,6 +995,24 @@ describe('NavigationBar', () => {
...
+      await openCollapsedMobileMenu(container);
+      const itemsRegion = await waitForMobilePanelPosition(container);
+      const toggle = container.querySelector('#toggle-btn') as HTMLButtonElement;
+      const enabledItem = itemsRegion.querySelector('[data-key="enabled"]');
+
+      toggle.focus();
+      await fireEvent.keyDown(toggle, { key: 'Tab' });
+      expect(document.activeElement).toBe(enabledItem);
+    });
+  });
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/navigation-bar/navigation-bar.test.ts:1127]: Implemented in d75d3d1e2. The pending focus flag is now reactive and has a regression-safe target fallback; sequential and arrow-focused tabindex=-1 boundary cases now bridge according to their logical edge. Focused NavigationBar tests, typecheck, lint, and format all pass.

File: packages/components/src/components/navigation-bar/navigation-bar.svelte [hunk 2]
```diff
@@ -112,11 +114,15 @@
...
+      pendingTabFocus = false;
+      pendingTabFocusTarget = null;
+    }
     if (!pendingTabFocus || !anchoredItems.positionReady) return;
     pendingTabFocus = false;
-    const firstItem = getNavigationItems().find(isEnabledNavigationItem);
-    firstItem?.focus();
+    const pendingTarget = pendingTabFocusTarget;
+    pendingTabFocusTarget = null;
+    queueMicrotask(() => (pendingTarget ?? getToggleTabTarget())?.focus());
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/navigation-bar/navigation-bar.svelte]: Implemented in d75d3d1e2. The pending focus flag is now reactive and has a regression-safe target fallback; sequential and arrow-focused tabindex=-1 boundary cases now bridge according to their logical edge. Focused NavigationBar tests, typecheck, lint, and format all pass.

File: packages/components/src/components/navigation-bar/navigation-bar.svelte [hunk 3]
```diff
@@ -343,7 +358,7 @@
   function bridgePortaledPanelTab(event: KeyboardEvent, navigationItem: HTMLElement): boolean {
     if (!anchoredItems.positionReady) return false;
 
-    const enabledItems = getNavigationItems().filter(isEnabledNavigationItem);
+    const enabledItems = getSequentialNavigationItems();
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/navigation-bar/navigation-bar.svelte:365]: Implemented in d75d3d1e2. The pending focus flag is now reactive and has a regression-safe target fallback; sequential and arrow-focused tabindex=-1 boundary cases now bridge according to their logical edge. Focused NavigationBar tests, typecheck, lint, and format all pass.

File: packages/components/src/components/navigation-bar/navigation-bar.svelte (hunk 2 above)
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/navigation-bar/navigation-bar.svelte]: Implemented in c5e1d3860. Added a failing regression for pending Tab with no sequential navigation item, then made the pending callback fall through to the same next logical focus target used after positioning (actions or following control). NavigationBar tests, typecheck, lint, format, component, and export checks pass.

File: packages/components/src/components/navigation-bar/navigation-bar.svelte [hunk 4]
```diff
@@ -343,18 +361,25 @@
   function bridgePortaledPanelTab(event: KeyboardEvent, navigationItem: HTMLElement): boolean {
     if (!anchoredItems.positionReady) return false;
 
-    const enabledItems = getNavigationItems().filter(isEnabledNavigationItem);
-    if (enabledItems.length === 0) return false;
+    const enabledItems = getSequentialNavigationItems();
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/navigation-bar/navigation-bar.svelte:365]: Implemented in d0b59ad50bada386a59237aa7d458403b9e42b99. The portal bridge now considers every sequential focus target in the items region, including inline controls, for the forward and reverse boundaries. Navigation items with tabindex=-1 remain excluded from initial sequential entry, while arrow-focused negative items retain their boundary fallback. Added a regression covering an inline control between the final sequential navigation item and a tabindex=-1 item.

[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/navigation-bar/navigation-bar.svelte:365]: Follow-up 1310ccbfd0900c4e6ec8d5146012102135b98955 tightens the fallback further: arrow-focused tabindex=-1 navigation items only bridge when no sequential focus target exists beyond them in the requested direction, so inline controls are never skipped.

File: packages/components/src/components/navigation-bar/navigation-bar.svelte [hunk 5]
```diff
@@ -343,18 +361,32 @@
...
       return true;
     }
 
-    if (!event.shiftKey && navigationItem === enabledItems.at(-1)) {
+    if (
+      !event.shiftKey &&
+      (navigationItem === sequentialItems.at(-1) ||
+        (!isSequentialTarget &&
+          (navigationItem === enabledItems.at(-1) ||
+            navigationItem === logicalEnabledItems.at(-1))))
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/navigation-bar/navigation-bar.svelte:396]: Fixed in 1310ccbfd0900c4e6ec8d5146012102135b98955. The non-sequential fallback now checks directional DOM order: it only bridges an arrow-focused tabindex=-1 item when no sequential target exists before/after it in the requested direction. Sequential inline controls therefore remain in the natural Tab path.

**Comment by stevekinney** [comment]
@copilot review
@codex review

Please review the exact current head `14db6c2e118cb28feca928953984f7d2da51e151` after the rebase onto `92d17da743f17902a87645cd5c92b8b0ce35e4c4`.
